### ISSUE TITLE
[REQ] option to store a "diff all" backup in a txt file on the SD card

### ISSUE THREAD

**Issue by b14ckyy** [iNavFlight/inav#8205]
## Current Behavior
Right now only Blackbox logs are stored on SD cards. Config backups have to be saved manually and flight stats have never a current backup if not done manually.

## Desired Behavior
Everytime the plane flew and is disarmed it should store a backup of the current config (a diff all would be enough) to the SD card in a dedicated file. This file will be overwritten everytime so only the latest config is saved to the SD card. 

## Suggested Solution
I see multiple options how this could be implemented. First the following criteria have to be met: 
- SD card available (or on board flash if possible as well, as it is a independent flash storage)
- Blackbox logging is enabled (although this would require storage reservation with on board flash as it will be filled up fast) 

When to make the backup? Multiple options (only one should be selected)
- on every disarm if Blackbox is enabled (although this might be overkill as the config won't change on disarm on all setups)
- Every time a save command is given (would backup the config every time when stats or auto trim is saved or if save command is given by pilot after landing)
- Only manually with stick save command (less writing but pilot has to think about it regularly) 

## Who does this impact? Who is this for?
Basically any pilot. Especially if people enable flight stats to have recordings about their flight time and distance on their crafts, this will ensure to keep these recordings. This could also open possibilities to revert back to an old config by reading it from SD if something goes wrong (like a config corruption) in the field. Also a damaged FC could be replaced and always have a current config ready if no backup was done on PC (as long it is on SD and not on board flash)

## Additional context
none

**Comment by b14ckyy** [comment]
@digitalentity Might be an alternative to your idea with a binary config as well?
